pkg/ebpf/controlplane/controller.go

```diff
@@ -169,16 +222,298 @@ func (p *Controller) processCgroupMkdir(args []trace.Argument) error {
 	return nil
 }
 
-func (p *Controller) processCgroupRmdir(args []trace.Argument) error {
+func (ctrl *Controller) processCgroupRmdir(args []trace.Argument) error {
 	cgroupId, err := parse.ArgVal[uint64](args, "cgroup_id")
 	if err != nil {
-		return errfmt.Errorf("error parsing cgroup_rmdir args: %v", err)
+		return errfmt.Errorf("error parsing cgroup_rmdir signal args: %v", err)
 	}
-
 	hId, err := parse.ArgVal[uint32](args, "hierarchy_id")
 	if err != nil {
-		return errfmt.Errorf("error parsing cgroup_rmdir args: %v", err)
+		return errfmt.Errorf("error parsing cgroup_rmdir signal args: %v", err)
+	}
+	ctrl.cgroupManager.CgroupRemove(cgroupId, hId)
+
+	return nil
+}
+
+//
+// SchedProcessFork, SchedProcessExec and SchedProcessExit
+//
+
+const SchedProcessForkAmountOfArgs = 10
+
+// processScheduleProcessFork parses the arguments of the SchedProcessFork signal and feeds the
+// process tree with the fork event.
+func (ctrl *Controller) processSchedProcessFork(args []trace.Argument) error {
+	var err error
+
+	var parentTid, parentNsTid, parentPid, parentNsPid int32
+	var childTid, childNsTid, childPid, childNsPid int32
+	var parentStartTime, childStartTime uint64
+
+	if len(args) != SchedProcessForkAmountOfArgs {
+		err = errfmt.Errorf("got %d args instead of %d", len(args), SchedProcessForkAmountOfArgs)
+		goto failed
```

Comment:
As I see this is only an error return, everywhere else we just do `return err` in tracee, so this `goto` is rather jarring stylistically. 
If there will be in the future some additional error handling before returning the error inside these 3 functions, then I guess it would be ok to keep (was this your prediction?), but otherwise let's stick to previous convention.

Reply:
Yep I know, but due to the amount of cars it got too repetitive and bigger lines and this was easier to read. Any other suggestion for readability?

Reply:
Well, the long error is only here. Other `goto` are near places where `return err` suffices. So there's not an issue of big lines really and doing the if err!=nil dance is just standard go (for better or worse).

Reply:
I'd say for only one error it's a bit redundant but yes that would also shorten this particular line (and the 2 other in the parallel functions).

Reply:
Yep will do then.